mindmeld/components/question_answerer.py

```
@@ -304,18 +404,54 @@ def transform(doc):
                     app_namespace,
                 )
 
+        def _generate_mapping_data(embedder_model, embedding_fields):
+            # generates a dictionary with any metadata needed to create the mapping"
+            if not embedder_model:
+                return {}
+            MAX_ES_VECTOR_LEN = 2048
+            embedding_properties = []
+            mapping_data = {"embedding_properties": embedding_properties}
+
+            dims = len(embedder_model.get_encodings(["encoding"])[0])
+            if dims > MAX_ES_VECTOR_LEN:
+                logger.error(
+                    "Vectors in ElasticSearch must be less than size: %d",
+                    MAX_ES_VECTOR_LEN,
+                )
+            for field in embedding_fields:
+                embedding_properties.append(
+                    {"field": field + EMBEDDING_FIELD_STRING, "dims": dims}
+                )
+
+            return mapping_data
+
+        es_client = es_client or create_es_client(es_host)
+        if is_es_version_7(es_client):
+            mapping_data = _generate_mapping_data(embedder_model, embedding_fields)
+            qa_mapping = create_index_mapping(DEFAULT_ES_QA_MAPPING, mapping_data)
+        else:
+            if embedder_model:
```

Should we throw an error here instead of logging it?